soepy/python/shared/shared_auxiliary.py

```diff
@@ -23,61 +23,105 @@ def draw_disturbances(seed, shocks_cov, num_periods, num_draws):
 
     # Calculate total period utilities by multiplying U(.) component
-    utilities = calculate_total_utilities(model_params, consumption_utilities)
+    flow_utilities = calculate_total_utilities(model_params, consumption_utilities)
 
     # Return function output
-    return utilities, consumption_utilities, period_wages, wage_systematic
+    return flow_utilities, consumption_utilities, period_wages, wage_systematic
+
 
+def calculate_wage_systematic(model_params, states, covariates):
+    """Calculate systematic wages, i.e., wages net of shock, for all states."""
 
-def calculate_wage_systematic(model_params, educ_level, exp_p, exp_f):
-    """Calculate systematic wages, i.e., wages net of shock, for specified state."""
+    exp_p, exp_f = states[:, 3], states[:, 4]
+    educ_level = covariates
 
     # Construct wage components
-    gamma_0s = np.dot(educ_level, model_params.gamma_0s)
-    gamma_1s = np.dot(educ_level, model_params.gamma_1s)
-    period_exp_sum = exp_p * np.dot(educ_level, model_params.g_s) + exp_f
-    depreciation = 1 - np.dot(educ_level, model_params.delta_s)
+    gamma_0s = np.dot(educ_level, np.array(model_params.gamma_0s))
+    gamma_1s = np.dot(educ_level, np.array(model_params.gamma_1s))
+    period_exp_sum = exp_p * np.dot(educ_level, np.array(model_params.g_s)) + exp_f
+    depreciation = 1 - np.dot(educ_level, np.array(model_params.delta_s))
 
     # Calculate wage in the given state
     period_exp_total = period_exp_sum * depreciation + 1
     returns_to_exp = gamma_1s * period_exp_total
     wage_systematic = np.exp(gamma_0s) * returns_to_exp
 
     # Return function output
-    return wage_systematic  # This is a scalar, equal for all choices
+    # Dimension (num_states x 1): scalar equal for all choices
+    return wage_systematic
 
 
-def calculate_period_wages(model_params, wage_systematic, draws):
+def calculate_period_wages(model_params, states, wage_systematic, draws):
     """Calculate period wages for each choice including choice
     and period specific productivity shock.
     """
-
     # Take the exponential of the disturbances
     exp_draws = np.exp(draws)
 
+    shape = (states.shape[0], draws.shape[1], NUM_CHOICES)
+    period_wages = np.full(shape, np.nan)
+
     # Calculate choice specific wages including productivity shock
-    period_wages = wage_systematic * exp_draws
+    for state in range(period_wages.shape[0]):
+        period = states[state, 0]
+        period_wages[state, :, :] = wage_systematic[state] * exp_draws[period, :, :]
 
     # Ensure that the benefits are recorded as non-labor income in data frame
-    period_wages[0] = model_params.benefits
+    period_wages[:, :, 0] = model_params.benefits
 
     # Return function output
-    return (
-        period_wages
-    )  # This is a vector, difference between choices comes from disturbance term.
+    # Dimension (num_states x num_draws x num_choices)
```

Comment:
Please move these kinds of comments to the docstrings.

Reply:
This is already included in the docstring of the wrapper function calculate_utilities, so I will just delete it here, okay?